app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.kt

```diff
@@ -0,0 +1,193 @@
+        try {
+            val md = MessageDigest.getInstance("SHA1")
+            val publicKey = md.digest(c!!.encoded)
+            hexString = byte2HexFormatted(publicKey)
+        } catch (e: NoSuchAlgorithmException) {
+            ErrorActivity.reportError(APP, e, null, null,
+                    ErrorActivity.ErrorInfo.make(UserAction.SOMETHING_ELSE, "none",
+                            "Could not retrieve SHA1 key", R.string.app_ui_crash))
+        } catch (e: CertificateEncodingException) {
+            ErrorActivity.reportError(APP, e, null, null,
+                    ErrorActivity.ErrorInfo.make(UserAction.SOMETHING_ELSE, "none",
+                            "Could not retrieve SHA1 key", R.string.app_ui_crash))
+        }
+        return hexString
+    }
+
+private val isConnected: Boolean
+    get() = APP.getSystemService<ConnectivityManager>()?.activeNetworkInfo?.isConnected == true
+
+private var disposable: Disposable? = null
+
+val isGithubApk: Boolean
+    get() = certificateSHA1Fingerprint == GITHUB_APK_SHA1
+
+private fun byte2HexFormatted(arr: ByteArray): String {
+    val str = StringBuilder(arr.size * 2)
+    for (i in arr.indices) {
+        var h = Integer.toHexString(arr[i].toInt())
+        val l = h.length
+        if (l == 1) {
+            h = "0$h"
+        }
+        if (l > 2) {
+            h = h.substring(l - 2, l)
+        }
+        str.append(h.toUpperCase(Locale.ROOT))
+        if (i < arr.size - 1) {
+            str.append(':')
+        }
+    }
+    return str.toString()
+}
+
+/**
+ * Method to compare the current and latest available app version.
+ * If a newer version is available, we show the update notification.
+ *
+ * @param versionName Name of new version
+ * @param apkLocationUrl Url with the new apk
+ * @param versionCode Code of new version
+ */
+private fun compareAppVersionAndShowNotification(versionName: String, apkLocationUrl: String, versionCode: Int) {
+    val notificationId = 2000
+    if (BuildConfig.VERSION_CODE < versionCode) {
+
+        // A pending intent to open the apk location url in the browser.
+        val intent = Intent(Intent.ACTION_VIEW, apkLocationUrl.toUri())
+        val pendingIntent = PendingIntent.getActivity(APP, 0, intent, 0)
+        val notificationBuilder = NotificationCompat.Builder(APP, APP.getString(R.string.app_update_notification_channel_id))
+                .setSmallIcon(R.drawable.ic_newpipe_update)
+                .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+                .setContentIntent(pendingIntent)
+                .setAutoCancel(true)
+                .setContentTitle(APP.getString(R.string.app_update_notification_content_title))
+                .setContentText(APP.getString(R.string.app_update_notification_content_text) + " " + versionName)
+        val notificationManager = NotificationManagerCompat.from(APP)
+        notificationManager.notify(notificationId, notificationBuilder.build())
+    }
+}
+
+fun checkNewVersion() {
```

Comment:
I know this is not the purpose of this PR, but while you are at it could you remove `APP = App.getApp()` and have this method take the app as an argument? This is also like how all other utility classes do
Also for `isGithubApk`

Reply:
Sure.